test/client.js

```diff
@@ -301,4 +301,21 @@ describe('http', () => {
       });
     });
   });
+
+  test('should omit undefined JSON query parameter', (done) => {
+    Swagger('http://localhost:8000/sample-query-parameter-json.yaml')
+      .then((client) => {
+        expect(client).toBeTruthy();
+        expect(Object.keys(client.apis).length).toBe(1);
+        expect(client.apis.default).toBeTruthy();
+        expect(client.apis.default.test).toBeTruthy();
+
+        return client.apis.default.test({}).catch((err) => {
+          // Would have thrown before that
```

Comment:
```suggestion
```

Can we remove the comment? As this might be confusing for people reading the test in future? The reader will be concerned how the code works now, instead of how the code works before with the bug.